azkaban-web-server/src/main/java/azkaban/scheduler/QuartzScheduler.java

```diff
@@ -108,7 +108,11 @@ public void shutdown() {
     }
   }
 
-  public void unregisterJob(final String groupName) throws SchedulerException {
+  /**
+   * Unregister a job given the groupname. Since unregister might be called when
+   * concurrently removing projects, so synchronized is added to ensure thread safety.
+   */
+  public synchronized void unregisterJob(final String groupName) throws SchedulerException {
```

Comment:
Wondering if it is possible to add a parameter in this API, because jobname is also necessary to locate a quartz Job. Would suggest we should do the change ASAP before flow trigger is in production.

Reply:
we can do it in next PR.